pkg/config/config.go

```diff
@@ -201,31 +178,31 @@ func GetNodeConfig(cfgPath string, k0sVars constant.CfgVars) (*v1beta1.ClusterCo
 	if err != nil {
 		return nil, err
 	}
+	nodeConfig := cfg.GetBootstrappingConfig()
 	var etcdConfig *v1beta1.EtcdConfig
 	if cfg.Spec.Storage.Type == v1beta1.EtcdStorageType {
 		etcdConfig = &v1beta1.EtcdConfig{
 			PeerAddress: cfg.Spec.Storage.Etcd.PeerAddress,
 		}
+		nodeConfig.Spec.Storage.Etcd = etcdConfig
 	}
+	return nodeConfig, nil
+}
 
-	clusterSpec := &v1beta1.ClusterSpec{
-		API: cfg.Spec.API,
-		Storage: &v1beta1.StorageSpec{
-			Type: cfg.Spec.Storage.Type,
-			Etcd: etcdConfig,
-			Kine: cfg.Spec.Storage.Kine,
-		},
-		Network: &v1beta1.Network{
-			ServiceCIDR: cfg.Spec.Network.ServiceCIDR,
-			DualStack:   cfg.Spec.Network.DualStack,
-		},
-		Install: cfg.Spec.Install,
-	}
-	nodeConfig := &v1beta1.ClusterConfig{
-		ObjectMeta: cfg.ObjectMeta,
-		TypeMeta:   cfg.TypeMeta,
-		Spec:       clusterSpec,
-		Status:     cfg.Status,
+func getConfigReader() (io.Reader, error) {
+	switch CfgFile {
+	case "-":
+		return os.Stdin, nil
+	case "", constant.K0sConfigPathDefault:
```

Comment:
If I read this correctly, this means that if `--config $configpath` is used, but `$configpath` does not exist, we will get an error - *unless*  `$configpath` happens to be identical with `constant.K0sConfigPathDefault`.

So a missing `--config /etc/k0s/k0s.conf` will lead to error, while a missing `--config /etc/k0s/k0s.yaml` will not error but silently use the defaults.

Wouldn't it be better to return error if `--config /etc/k0s/k0s.yaml` is used but the file is missing?

```suggestion
	case "":
```

Reply:
> Wouldn't it be better to return error if `--config /etc/k0s/k0s.yaml` is used but the file is missing?

In my opinion, no - because the flag configuration for `--config` always defaults now to `/etc/k0s/k0s.yaml`, and that means that users who have not used any `--config` flag input (as was used before) will always fail. And that will hurt backwards compatibility.

Reply:
> that means that users who have not used any --config flag input (as was used before) will always fail.

What I mean is that when `--config` is used, it should always fail if the specified file cannot be read, regardless if it is the default config file or not.

When no `--config` is specified, it should try the default config path, but not fail if it does not exist. It should still fail if it fails to read for any other reason (permission denied for example).